packages/commercetools/helpers/src/index.ts

```diff
@@ -1,19 +1,23 @@
 import { UiMediaGalleryItem } from '@vue-storefront/interfaces'
 import { ProductVariant, Image, Category } from './types/GraphQL'
 
-export const getProductName = (product: ProductVariant): string => (product as any)._name
+export const getProductName = (product: ProductVariant): string => product ? (product as any)._name : ''
 
-export const getProductSlug = (product: ProductVariant): string => (product as any)._slug
+export const getProductSlug = (product: ProductVariant): string => product ? (product as any)._slug : ''
 
-export const getProductPrice = (product: ProductVariant): number => product.price.value.centAmount
+export const getProductPrice = (product: ProductVariant): number => product ? product.price.value.centAmount : null
 
-export const getProductGallery = (product: ProductVariant): UiMediaGalleryItem[] =>
-  product.images.map((image: Image) => ({
+export const getProductGallery = (product: ProductVariant): UiMediaGalleryItem[] => {
+  if (!product) {
+    return []
+  }
+
+  return product.images.map((image: Image) => ({
```

Comment:
Just an idea, but I'm not sure you'll like it:
```TS
export const getProductGallery = (product: ProductVariant): UiMediaGalleryItem[] => (product ? product.images : [])
  .map((image: Image)) => ({ ....other stuff })
```

Reply:
makes sense